src/gt4py/cartesian/gtc/debug/debug_codegen.py

```diff
@@ -0,0 +1,354 @@
+    def visit_Interval(self, interval: Interval, **kwargs):
+        if kwargs["direction"] == LoopOrder.BACKWARD:
+            return ",".join(
+                [
+                    self.visit(interval.end, **kwargs) + "- 1",
+                    self.visit(interval.start, **kwargs) + "- 1",
+                ]
+            )
+        else:
+            return ",".join(
+                [self.visit(interval.start, **kwargs), self.visit(interval.end, **kwargs)]
+            )
+
+    def visit_Temporary(self, temporary_declaration: Temporary, **kwargs) -> str:
+        field_extents = kwargs["field_extents"]
+        local_field_extent = field_extents[temporary_declaration.name]
+        i_padding: int = local_field_extent[0][1] - local_field_extent[0][0]
+        j_padding: int = local_field_extent[1][1] - local_field_extent[1][0]
+        shape: list[str] = [f"i_size + {i_padding}", f"j_size + {j_padding}", "k_size"]
+        data_dimensions: list[str] = [str(dim) for dim in temporary_declaration.data_dims]
+        shape = shape + data_dimensions
+        shape_decl = ", ".join(shape)
+        dtype: str = self.visit(temporary_declaration.dtype)
+        field_offset = tuple(-ext[0] for ext in local_field_extent)
+        offset = [str(off) for off in field_offset] + ["0"] * (
+            1 + len(temporary_declaration.data_dims)
+        )
+        return f"{temporary_declaration.name} = Field.empty(({shape_decl}), {dtype}, ({', '.join(offset)}))"
+
+    def visit_DataType(self, data_type: DataType, **_) -> str:
+        if data_type not in {DataType.BOOL}:
+            return f"np.{data_type.name.lower()}"
+        else:
+            return data_type.name.lower()
+
+    def visit_VariableKOffset(self, variable_k_offset: VariableKOffset, **_) -> str:
+        return f"i,j,k+int({self.visit(variable_k_offset.k)})"
+
+    def visit_CartesianOffset(self, cartesian_offset: CartesianOffset, **kwargs) -> str:
+        dimensions = kwargs["dimensions"]
+        return cartesian_offset.to_str(dimensions)
+
+    def visit_SymbolRef(self, symbol_ref: SymbolRef) -> str:
+        return symbol_ref
+
+    def visit_FieldAccess(self, field_access: FieldAccess, **_) -> str:
+        if str(field_access.name) in self.symbol_table:
+            dimensions = self.symbol_table[str(field_access.name)].dimensions
+        else:
+            dimensions = (True, True, True)
+
+        offset_str = self.visit(field_access.offset, dimensions=dimensions)
+
+        if field_access.data_index:
+            data_index_access = ",".join(
+                [self.visit(data_index) for data_index in field_access.data_index]
+            )
+            if offset_str == "":
+                full_string = field_access.name + f"[{data_index_access}]"
+            else:
+                full_string = field_access.name + "[" + offset_str + "," + data_index_access + "]"
+        else:
+            full_string = field_access.name + "[" + offset_str + "]"
+        return full_string
```

Comment:
how about using early return statements instead?

```suggestion
        if field_access.data_index:
            data_index_access = ",".join(
                [self.visit(data_index) for data_index in field_access.data_index]
            )
            if offset_str == "":
                return field_access.name + f"[{data_index_access}]"

            return field_access.name + "[" + offset_str + "," + data_index_access + "]"

        return field_access.name + "[" + offset_str + "]"
```